### ISSUE TITLE
pointwise_operator doesn't compose with make_fx

### ISSUE THREAD

**Issue by bertmaher** [pytorch/functorch#186]
The design we've been kicking around is to use pythonkey tracing to get the forward/backward expression for a user-defined pointwise operator, but pointwise_operator doesn't work with pythonkey-traced graphs.  See:
```
    def test_make_fx(self):
        def example(a, b):
            return a + b
        a = torch.ones(1)
        b = torch.ones(1)
        gm = make_fx(example)(a, b)
        gm.__name__ = "example_fx"  # get around trivial failure                                                                                                                                                                                                                                                                                                                               
        cm = pointwise_operator(gm)
        cm(a, b)
 ```

I think the root cause is that pointwise_operator tries to execute its argument with te.ExprHandle arguments, which works ok for operators that are overloaded by te.ExprHandle (+, -, *, sigmoid, etc.), but it doesn't know how to handle `torch.ops.aten.add`.

I suspect we need an NNCTensor that wraps an ExprHandle and knows how to translate torch API calls into operations on ExprHandle.
